### DISCUSSION TITLE
How to compile a dir to a single file for nodejs?

### DISCUSSION THREAD

**Discussion by wxs77577** [swc-project/swc#1474] [Q&A] [answered]
I'm building a nodejs typescript project.

`npx swc src -o dist.js && node dist.js`
It doesn't work because it uses `require('./user.js')` in dist file. But swc compiles all files to a single file (as `exports.User = xxx` ), so there is no `./user.js` , it should replace all required local files to `exports.xxx`

**Comment by kdy1** [comment, marked as answer]
You can use `spack` to bundle files.
See https://swc.rs/docs/spack-basic for `spack.config.js`

**Comment by electrovir** [comment]
Usage docs are here: https://swc.rs/docs/usage/bundling
